framework/src/media/codec/wav/wav_decoder_api.cpp

```diff
@@ -207,10 +207,13 @@ bool wav_get_frame(rbstream_p mFp, ssize_t *offset, void *dec_mem, void *buffer,
 		n = _source_read_at(mFp, _offset + 4, sizeBuf, sizeof(sizeBuf));
 		RETURN_VAL_IF_FAIL(((size_t)n == sizeof(sizeBuf)), false);
 		uint32_t chunkSize = _le_num_at(&sizeBuf[0], 4);
-		uint8_t fmtChunk[8 + chunkSize];
-		n = _source_read_at(mFp, _offset, fmtChunk, sizeof(fmtChunk));
-		RETURN_VAL_IF_FAIL(((size_t)n == sizeof(fmtChunk)), false);
-		RETURN_VAL_IF_FAIL((_format_parse((void *)fmtChunk, sizeof(fmtChunk), ext)), false);
+		uint32_t fmtChunkSize = 8 + chunkSize;
+		uint8_t *fmtChunk = new uint8_t[fmtChunkSize];
```

Comment:
How about using `malloc`&`free` instead of `new`&`delete`? to keep code in C style.
(I will rename this file to **wav_decoder_api.c**)

Reply:
I raised PR #3585

Reply:
Fixed it